Source/Urho3D/Scene/ShakeComponent.h

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright (c) 2023-2023 the rbfx project.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+#include "Urho3D/Math/PerlinNoise.h"
+#include "Urho3D/Scene/LogicComponent.h"
+
+namespace Urho3D
+{
+
+/// Component that randomly displaces the node around (0,0,0) coordinate.
+/// Perfect for camera shake effect.
+///
+/// Math for Game Programmers: Juicing Your Cameras With Math by Squirrel Eiserloh
+/// https://www.gdcvault.com/play/1033548/Math-for-Game-Programmers-Juicing
+///
+class URHO3D_API ShakeComponent : public LogicComponent
```

Comment:
This component is dead code now -- untested in the engine in any way. If it breaks, no one will know.
Can we do sample?

Reply:
Sure but making a new dedicated sample for this feels like too much. Or is it? Shall I add to some existing sample instead?